### DISCUSSION TITLE
Trouble producing some probability stats in EnsembleStat

### DISCUSSION THREAD

**Discussion by wuxx66** [dtcenter/METplus#2849] [Configuration]
I have defined "ENSEMBLE_STAT_PROB_CAT_THRESH =  >=CDP68, >=CDP87, >=CDP95" in my config file and tried to produce PCT, PSTD and so on, but the task failed. The task runs successfully for  "ENSEMBLE_STAT_PROB_CAT_THRESH = , >=CDP87, >=CDP95" but failed only for  "ENSEMBLE_STAT_PROB_CAT_THRESH =  >=CDP68" 

DEBUG 4: For threshold >=CDP68, using a constant climatological probability value of 0.32.
DEBUG 3: Using 3546 of 6480 pairs for climatology bin number 10.
DEBUG 4: For threshold >=CDP68, using a constant climatological probability value of 0.32.
ERROR  :
ERROR  :   AsciiTable::rc_to_n() -> range check error ...
ERROR  :                            (Nrows, Ncols) = (67, 58)
ERROR  :                            (r, c) = (67, 0)
ERROR  :

Could you help me have a look what I did wrong?

I have uploaded the config file and log file into "wu_data" folder.

Thank you

Xiaoxi

**Comment by JohnHalleyGotway** [comment]
@wuxx66 thanks for sending along your sample configuration file and METplus log output file. I see that you're getting a runtime error from Ensemble-Stat, and you increased the logging level verbosity to try to debug:
```
LOG_ENSEMBLE_STAT_VERBOSITY = 5
```
And that's great. I'll note that it is also useful to tell the METplus wrappers to log more details too, which is done by increasing the [LOG_LEVEL](https://metplus.readthedocs.io/en/main_v5.1/Users_Guide/systemconfiguration.html#log-level) setting from `INFO` to `DEBUG`:
```
LOG_LEVEL = DEBUG
```
I spent a while trying to replicate this failure using sample data locally, but wasn't able to do so. I note that I've definitely seen this TYPE of error before. Based on the config file contents, the ensemble_stat tool figures out how many rows and columns of ASCII output it will write. In particular, this [n_txt_row()](https://github.com/dtcenter/MET/blob/c5a2155ada4773b2d3c7f50d5dea8f1dff791153/src/tools/core/ensemble_stat/ensemble_stat_conf_info.cc#L1056) function counts of the number of rows it'll need for a given "verification task".

The range check error you're seeing means that the we didn't do that counting correctly based on your configuration settings. It ran out of room when trying to write the 67-th row.
```
ERROR : AsciiTable::rc_to_n() -> range check error ...
ERROR : (Nrows, Ncols) = (67, 58)
ERROR : (r, c) = (67, 0)
```
Can you please re-run with `LOG_LEVEL = DEBUG` and send me the resulting log file? That will help me confirm that actual configuration options being passed to MET's ensemble_stat tool.

**Reply by wuxx66** [reply]
Hi John,
Thank you for helping me with this issue. I have added "LOG_LEVEL = DEBUG" in the config file and sent the new log file to the ftp site.

I have done several more tests. I found if the config file has only 1 CDP threshold or any 2 CDP thresholds, the task will run successfully, but if the config file has 3 CDP together, the task will fail.

For example,
If I set 
ENSEMBLE_STAT_PROB_CAT_THRESH =  >=CDP68
It works. It will also work for >=CDP87 or>=CDP95.

If I set 
ENSEMBLE_STAT_PROB_CAT_THRESH =  >=CDP87, >=CDP95 
It works too. It also works for any 2 combinations.

If I set
ENSEMBLE_STAT_PROB_CAT_THRESH =  >=CDP68, >=CDP87, >=CDP95
The task fails.

Hope this can give you some hints,

Thank you

Xiaoxi

**Reply by wuxx66** [reply]
Hi John @JohnHalleyGotway 
It has been a while since I raised the CDP issue. Just want to follow up if there is any progress on it.

Thank you

Xiaoxi
